### PR TITLE
Added rtl toggle button

### PR DESCRIPTION
**Pull request by ibraude**
## Description

I have added a direction toggle switch for the MarkdownPreview and CodeEditor. (It's located at the top right, next to the Toggle Mode switch button)
This is so notes can be taken and read comfortably when using right-to-left languages like Hebrew or Arabic.

Here is the Boostnote welcome page in the LTR direction: 
image

Here is the Boostnote welcome page in Hebrew with the RTL direction: 
image

I have also added a hotkey for direction toggle: 
image

## Issue fixed
This fixes the open issue "Please add RTL support" #846 
I believe this is a simple solution that answers the need for people using both ltr and rtl languages regularly. 

## Type of changes

- :radio_button: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :white_circle: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

All tests pass except for the test: ` tests/lib/markdown-toc-generator-test.js`
I think that the test throws an error because it recognizes the difference between the rtl to ltr as seen in the image below when infact it is the desired result. 

image

Hope this helps.

Cheers

### CODE REVIEW

File: browser/main/Detail/ToggleDirectionButton.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import React from 'react'
+import CSSModules from 'browser/lib/CSSModules'
+import styles from './ToggleDirectionButton.styl'
+import i18n from 'browser/lib/i18n'
+
+const ToggleDirectionButton = ({
+  onClick, editorDirection
+}) => (
+  <div styleName='control-toggleModeButton'>
+    <div styleName={editorDirection ? 'active' : undefined} onClick={() => onClick()}>
```
Comment (ZeroX-DG) [inline browser/main/Detail/ToggleDirectionButton.js]: `editorDirection` is quite hard to understand, can you assign it to a more meaningful variable name such as `isRTL` or something?

**Comment by ZeroX-DG** [comment]
@ibraude ping!

**Comment by ibraude** [comment]
@ZeroX-DG Sorry for the delay

**Comment by ZeroX-DG** [comment]
@ibraude can you fix the conflict before I officially approve it?

File: browser/components/MarkdownPreview.js [hunk 2]
```diff
@@ -101,11 +102,15 @@ ${markdownStyle}
 body {
   font-family: '${fontFamily.join("','")}';
   font-size: ${fontSize}px;
+
   ${scrollPastEnd ? `
     padding-bottom: 90vh;
     box-sizing: border-box;
     `
     : ''}
+  ${optimizeOverflowScroll ? 'height: 100%;' : ''}
```
Comment (ZeroX-DG) [inline browser/components/MarkdownPreview.js]: This `optimizeOverflowScroll` variable is undefined can you fix it please

**Pull request merged**